xarray/core/variable.py

```diff
@@ -1099,24 +1103,21 @@ def _shift_one_dim(self, dim, count, fill_value=dtypes.NA):
         else:
             dtype = self.dtype
 
-        shape = list(self.shape)
-        shape[axis] = min(abs(count), shape[axis])
+        width = min(abs(count), self.shape[axis])
+        dim_pad = (width, 0) if count >= 0 else (0, width)
+        pads = [(0, 0) if d != dim else dim_pad for d in self.dims]
 
         if isinstance(trimmed_data, dask_array_type):
-            chunks = list(trimmed_data.chunks)
-            chunks[axis] = (shape[axis],)
-            full = functools.partial(da.full, chunks=chunks)
+            pad_func = da.pad
         else:
-            full = np.full
-
-        filler = full(shape, fill_value, dtype=dtype)
+            pad_func = np.pad
```

Comment:
We try to consolidate all our duck typing aliases into duck_array_ops.py. Can you make a `pad` function there instead?

Reply:
sure